src/main/java/common/tileentities/GTMTE_LapotronicSuperCapacitor.java

```diff
@@ -295,10 +297,34 @@ public boolean checkMachine(IGregTechTileEntity thisController, ItemStack guiSlo
 		final int colourCorrectedMeta = firstGlassMeta > 5 ? 0 : firstGlassMeta;
 		for(int highestCapacitor = capacitors.length - 1; highestCapacitor >= 0; highestCapacitor--){
 			if(capacitors[highestCapacitor] > 0){
-				if(colourCorrectedMeta < highestCapacitor){
-					formationChecklist = false;
-				}
-				break;
+				switch (highestCapacitor) {
```

Comment:
You shouldn't use switches in functions. Wrap them please in a higher order function.